pallets/pips/src/lib.rs

```diff
@@ -1341,23 +1343,23 @@ impl<T: Config> Module<T> {
         id: PipId,
         new_state: ProposalState,
     ) -> ProposalState {
-        <Proposals<T>>::mutate(id, |proposal| {
-            if let Some(ref mut proposal) = proposal {
-                // Decrement active count, if the `new_state` is not active.
-                if !Self::is_active(new_state) {
-                    Self::decrement_count_if_active(proposal.state);
+        <ProposalStates>::mutate(id, |proposal_state| {
+            // Decrement active count, if the `new_state` is not active.
+            if !Self::is_active(new_state) {
+                if let Some(proposal_state) = proposal_state {
+                    Self::decrement_count_if_active(*proposal_state);
                 }
-                proposal.state = new_state;
             }
+            *proposal_state = Some(new_state);
```

Comment:
This changed the logic, the two nested `if` statements are in a different order.  Also it will always set the PIP state, even if the old PIP state was `None`, which is different that what it would do before.

Please keep the old logic and only change storage from `Proposals` to `ProposalStates`.